kernel/window_manager/src/lib.rs

```diff
@@ -684,30 +684,26 @@ fn window_manager_loop(
                     keyboard_handle_application(key_input)?;
                 }
                 Event::MouseMovementEvent(ref mouse_event) => {
-                    // mouse::mouse_to_print(&mouse_event);
-                    let mouse_displacement = &mouse_event.displacement;
-                    let mut x = (mouse_displacement.x as i8) as isize;
-                    let mut y = (mouse_displacement.y as i8) as isize;
-                    // need to combine mouse events if there pending a lot
+                    //as isize probably because adding up mouse movements will overflow?
+                    let mut x = mouse_event.mousemove.x_movement as isize;
+                    let mut y = mouse_event.mousemove.y_movement as isize;
 
+                    // need to combine mouse events if there pending a lot
                     while let Some(next_event) = mouse_consumer.pop() {
                         match next_event {
                             Event::MouseMovementEvent(ref next_mouse_event) => {
-                                if next_mouse_event.mousemove.scrolling_up
-                                    == mouse_event.mousemove.scrolling_up
-                                    && next_mouse_event.mousemove.scrolling_down
-                                        == mouse_event.mousemove.scrolling_down
+                                if next_mouse_event.mousemove.scroll_movement
+                                    == mouse_event.mousemove.scroll_movement
                                     && next_mouse_event.buttonact.left_button_hold
                                         == mouse_event.buttonact.left_button_hold
                                     && next_mouse_event.buttonact.right_button_hold
                                         == mouse_event.buttonact.right_button_hold
                                     && next_mouse_event.buttonact.fourth_button_hold
                                         == mouse_event.buttonact.fourth_button_hold
                                     && next_mouse_event.buttonact.fifth_button_hold
-                                        == mouse_event.buttonact.fifth_button_hold
-                                {
-                                    x += (next_mouse_event.displacement.x as i8) as isize;
-                                    y += (next_mouse_event.displacement.y as i8) as isize;
+                                        == mouse_event.buttonact.fifth_button_hold {
+                                    x += next_mouse_event.mousemove.x_movement as isize;
+                                    y += next_mouse_event.mousemove.y_movement as isize;
```

Comment:
above this, you mentioned being concerned about overflow. I agree, and have recently been auditing other parts of the Theseus codebase that use arithmetic operations without handling overflow (e.g., #663).

Thus, we should probably use either `wrapping_add` or `saturating_add` here, whichever is appropriate according to the semantics of these values.

Reply:
or, if you want to disallow overflow, `checked_add` is also available. But that likely isn't the correct choice here.

Reply:
thank you! forgot about those